src/main/java/com/epam/ta/reportportal/core/events/handler/subscriber/impl/LaunchNotificationSubscriber.java

```diff
@@ -0,0 +1,207 @@
+import com.epam.ta.reportportal.entity.enums.SendCase;
+import com.epam.ta.reportportal.entity.enums.StatusEnum;
+import com.epam.ta.reportportal.entity.integration.Integration;
+import com.epam.ta.reportportal.entity.launch.Launch;
+import com.epam.ta.reportportal.entity.project.Project;
+import com.epam.ta.reportportal.entity.project.ProjectUtils;
+import com.epam.ta.reportportal.entity.project.email.LaunchAttributeRule;
+import com.epam.ta.reportportal.entity.project.email.SenderCase;
+import com.epam.ta.reportportal.entity.user.User;
+import com.epam.ta.reportportal.exception.ReportPortalException;
+import com.epam.ta.reportportal.util.email.EmailService;
+import com.epam.ta.reportportal.util.email.MailServiceFactory;
+import com.epam.ta.reportportal.ws.converter.converters.NotificationConfigConverter;
+import com.epam.ta.reportportal.ws.model.ErrorType;
+import com.epam.ta.reportportal.ws.model.attribute.ItemAttributeResource;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.BooleanUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.epam.ta.reportportal.core.statistics.StatisticsHelper.extractStatisticsCount;
+import static com.epam.ta.reportportal.dao.constant.WidgetContentRepositoryConstants.*;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+@Service
+public class LaunchNotificationSubscriber implements LaunchFinishedEventSubscriber {
+
+	public static final Logger LOGGER = LoggerFactory.getLogger(LaunchNotificationSubscriber.class);
+
+	private final GetIntegrationHandler getIntegrationHandler;
+	private final MailServiceFactory mailServiceFactory;
+	private final LaunchRepository launchRepository;
+	private final UserRepository userRepository;
+
+	@Autowired
+	public LaunchNotificationSubscriber(GetIntegrationHandler getIntegrationHandler, MailServiceFactory mailServiceFactory,
+			LaunchRepository launchRepository, UserRepository userRepository) {
+		this.getIntegrationHandler = getIntegrationHandler;
+		this.mailServiceFactory = mailServiceFactory;
+		this.launchRepository = launchRepository;
+		this.userRepository = userRepository;
+	}
+
+	@Override
+	public void handleEvent(LaunchFinishedEvent launchFinishedEvent, Project project, Launch launch) {
+
+		boolean isNotificationsEnabled = BooleanUtils.toBoolean(ProjectUtils.getConfigParameters(project.getProjectAttributes())
+				.get(ProjectAttributeEnum.NOTIFICATIONS_ENABLED.getAttribute()));
+
+		if (isNotificationsEnabled) {
+			Integration emailIntegration = getIntegrationHandler.getEnabledByProjectIdOrGlobalAndIntegrationGroup(project.getId(),
```

Comment:
we suppose this integration to be not only email, but slack/etc
Can we introduce some abstraction or at least make sure that this integration is really email